docs/src/get-started/concepts.rst

```diff
@@ -0,0 +1,239 @@
+.. _userdoc-core-concepts:
+
+Core concepts
+=============
+
+This page explains the main types and core concepts you need to understand to
+use equistore, without tying it to how a specific library is using equistore to
+store it's own data.
+
+.. py:currentmodule:: equistore
+
+.. _concept-TensorMap:
+
+TensorMap
+^^^^^^^^^
+
+The core type of equistore is the :py:class:`TensorMap`: a high dimensional
+block-sparse tensor containing both data and metadata. A TensorMap contains a
+list of block (represented as :ref:`concept-TensorBlock`), each associated with
+a key. The set of all keys is stored in a :ref:`concept-Labels` object
+(described below), which can be accessed in Python via
+:py:attr:`TensorMap.keys`.
+
+
+The keys can contain multiple dimensions (in the illustration below we have two
+dimensions named ``lambda`` and ``sigma``), and each entry in the keys has one
+integer value for each dimension. Here for example, the first block is
+associated with ``lambda = 0`` and ``sigma = 0``, while the second block is
+associated with ``lambda = 0`` and ``sigma = 1``, and so on.
+
+.. figure:: ../../static/images/TensorMap.*
+    :width: 600px
+    :align: center
+
+    Illustration of an equistore TensorMap object, made of a set of keys and
+    associated :ref:`concept-TensorBlock`.
+
+
+Different key dimensions can have different purposes, but some typical keys
+dimensions you'll encounter when working with atomistic data are the following:
+
+- **atomic species dimensions**: when using a one-hot encoding of different
+  atomic species (or atomic elements) in a structure, the resulting data is
+  sparse, containing implicit zeros if a given species is not present in a
+  structure. This is the case of the ``species_center`` and various
+  ``species_neighbor`` key dimensions produced by `rascaline`_.
```

Comment:
```suggestion
- **atomic species**: for different elements, we often encode their
  using one-hot encodings. However, when a species is not present
  in a dataset, traditional numpy-like formats will be sparse, with 
  implicit zeros wherever the species is not present.
  This is the case of the ``species_center`` and various
  ``species_neighbor`` key dimensions produced by `rascaline`_.
```

Reply:
> However, when a species is not present in a dataset, traditional numpy-like formats will be sparse, with implicit zeros wherever the species is not present.

that's not the case: numpy-like storage would have to store **explicit** zeros